### ISSUE TITLE
Inspector not running code in `@tool` scripts until restart

### ISSUE THREAD

**Issue by NumbuhFour** [godotengine/godot#66381]
___

***Bugsquad note:** This issue has been confirmed several times already. No need to confirm it further.*

___ 

### Godot version

4.0 beta 1

### System information

Windows 10, Vulkan desktop

### Issue description

No script marked as `@tool` appears to be executing when it is expected by the inspector. Examples I have found are that the set/get functions of properties are not being called and the parse_begin / parse_end functions of inspector plugins are not being called.

### Steps to reproduce

1. Create a new Node3D scene
2. Attach a script with the following code:
```@tool
extends Node3D

@export var tester:int:
    get:
        return 100
    set(value):
        print("NEW VALUE ", value)
        tester = value/2
```
3. Select your node and change the `tester` property in the inspector:

**Expected:**
Value of `tester` to be set to half the input value. "NEW VALUE" should appear in the output log

**Result:**
Value of `tester` is set to exactly the input value. No change in output log

### Minimal reproduction project

[ToolBug.zip](https://github.com/godotengine/godot/files/9639750/ToolBug.zip)

**Comment by NumbuhFour** [comment]
Through further experimentation, I have found that removing the script from the node and adding it back gets it to work

**Comment by dmaz** [comment]
I was just about to post this for setters.
while work around for the `int` setter above seems to work, it doesn't seem to even try for things that are in the tree like a `Node3D`. 
```
@tool
extends MeshInstance3D

@export var vint:int:
	set(v):	print("int ", v)

@export var vVector3:Vector3:
	set(v): print('Vector3 ',v)

@export var vNode3D:Node3D:
	set(v): print('Node3D ',v) ; vNode3D = v

@export var vResource:Resource:
	set(v): print('Resource ',v) ; vResource = v
```
everything but the Node3D doesn't ever print whether you just add the scene or even change the setting.  

on run it does, though the Node3D setter fires later in order after other scripts have initialized.

**Comment by Calinou** [comment]
Related to https://github.com/godotengine/godot/issues/65916 (possible duplicate?).

**Comment by hsandt** [comment]
> Through further experimentation, I have found that removing the script from the node and adding it back gets it to work

On my side, with a reference (Marker2D) field, reopening the scene or re-adding the script doesn't fix the issue. The setter simply seems ignored. Even assert(false) will show nothing.

Since the issue only happens with get/set and assigning field in the inspector, should we rename the issue to something like "Inspector not running property get/set when reading/assigning field"?

**Comment by jmdejong** [comment]
In my case the set action does work while the game itself is running and that it does not work when the game is not running

```
@tool
extends Node2D

@export var seed : int = 0:
	set(new_seed):
		seed = new_seed
		print("the seed is ", seed)
```

**Comment by Mietzsche81** [comment]
Also seeing this issue: I'm **unable to run scripts in the editor via @tool**. 

Following the [official docs' tutorial on @tool](https://docs.godotengine.org/en/stable/tutorials/plugins/running_code_in_the_editor.html) verbatim doesn't work.

**Comment by Mietzsche81** [comment]
IMO, I think this is a high priority feature to get working. Without editor scripting working, procedural generation is a total "guess and check" operation.

**Comment by Mietzsche81** [comment]
@AThousandShips  
> is it different from what's already mentioned here?

Yes, to clarify: I am _not_ able to run scripts in the editor in general. `v4.0.2.stable.official.7a0977ce2`  on Windows 10.

I have tried closing/reopening the scene and still can't even get the tutorial's rotation behavior to work.

**Comment by AThousandShips** [comment]
Have you restarted the editor?

**Comment by AThousandShips** [comment]
This is unfortunately a current limitation which I'm not sure why it is, nor is it documented, can't find an issue tracking it either

**Comment by Mietzsche81** [comment]
@AThousandShips huge thanks, that did the trick. 

In case it's helpful to anyone:

**When adding/removing `@tool` from a script or attaching/detaching a `@tool`ed script to a scene, you must restart the editor to add/remove interactivity in the editor**

After restarting, I'm now able to edit the script and see the behavior change immediately without having to restart the editor. But if I create a new scene and attach the same script to it, I have to restart the editor again to get the behavior into the process loop.

**Comment by Zireael07** [comment]
@AThousandShips lots of people been running into this, really ought to be documented !!!

**Comment by AThousandShips** [comment]
Agreed, if an issue over it can't be find someone please open one (I'm preoccupied at the moment)

**Comment by dalexeev** [comment]
See also:

* #81972
  > Tested on 3.5.2
  > The behavior was instantaneous and worked fine. It didn't require reload of scene or Project
* #82443
  > Reopening the scene also works, not just restarting godot entirely.

**Comment by nanodeath** [comment]
> does this still happen in 4.1.3 or 4.2 betas

@YuriSizov I believe I can confirm this is still happening in 4.2-rc.2, though the behavior I'm seeing might not quite be the same. Also I'm using C#.

1. Create main scene.
2. Add another scene ("secondary scene") as a child of the main scene.
3. To a child of the secondary scene, add a script with the [Tool] annotation.

This Tool script will _only run_ when the **main** scene is opened, and not the secondary scene. Restarting Godot enables it to run in both scenes.

**Comment by RedGlow** [comment]
Can confirm the bug still exists on 4.2 also with GDScript.

**Comment by mournguard** [comment]
Bump happening to me as well on v4.3.dev5.official [89f70e98d]
